cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java

```diff
@@ -0,0 +1,344 @@
+        // if there are no more balos to push.
+        if (balos.size() == 0) {
+            return;
+        }
+
+        // go through the dependencies of balos and see if they are available in remote repository. if they are
+        // available remove them from the list.
+        for (List<PackageDescriptor.Dependency> deps : balos.values()) {
+            Iterator<PackageDescriptor.Dependency> depsIterator = deps.iterator();
+            while (depsIterator.hasNext()) {
+                PackageDescriptor.Dependency dep = depsIterator.next();
+                if (isDependencyAvailableInRemote(dep)) {
+                    depsIterator.remove();
+                }
+
+                if ("ballerina".equals(dep.org().toString()) || "ballerinax".equals(dep.org().toString())) {
+                    depsIterator.remove();
+                }
+            }
+        }
+
+        // check if there are balos where their dependencies are already available in remote repository
+        Optional<List<PackageDescriptor.Dependency>> baloWithAllDependenciesAvailableInCentral = balos.values().stream()
+                .filter(List::isEmpty)
+                .findAny();
+
+        // if there isn't any balos where dependencies are resolved, then throw an error.
+        if (!baloWithAllDependenciesAvailableInCentral.isPresent()) {
+            Set<String> unresolvedDependencies = balos.values().stream()
+                    .flatMap(List::stream)
+                    .map(PackageDescriptor.Dependency::toString)
+                    .collect(Collectors.toSet());
+            throw createLauncherException("unable to find dependencies in remote repository: [" +
+                    String.join(", ", unresolvedDependencies) + "]");
+        }
+
+        // push all the modules where dependencies are available in remote repository and remove them from the map.
+        Iterator<Map.Entry<Path, List<PackageDescriptor.Dependency>>> iterator = balos.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<Path, List<PackageDescriptor.Dependency>> baloDeps = iterator.next();
+            if (baloDeps.getValue().isEmpty()) {
+                pushBaloToRemote(baloDeps.getKey());
+                iterator.remove();
+            }
+        }
+        recursivelyPushBalos(balos);
+    }
+
+    /**
+     * Push a balo file to remote repository.
+     *
+     * @param baloPath Path to the balo file.
+     */
+    private static void pushBaloToRemote(Path baloPath) {
+        Path baloFileName = baloPath.getFileName();
+        if (null != baloFileName) {
+            // Load BaloProject from balo path
+            BaloProject baloProject = BaloProject.loadProject(baloPath, null);
+            String name = baloProject.currentPackage().packageName().toString();
+
+            try {
+                CentralAPIClient client = new CentralAPIClient();
+                client.pushPackage(baloPath, baloProject);
```

Comment:
Why do we need to pass the balo project? cant we create it inside the client

Reply:
created inside client